integration-tests/tests/run-notebook.js

```diff
@@ -46,6 +46,9 @@ const testRunNotebookFn = _.flow(
   await findElement(page, clickable({ textContains: 'Running' }), { timeout: 10 * 60 * 1000 })
 
   const frame = await findIframe(page)
+  await findElement(frame, '//*[@title="Kernel Idle"]')
+  //Delay here to allow runtime to finish creating.
+  await delay(10000)
```

Comment:
What are your thoughts on correcting this without adding a delay?

Reply:
I would love if we didn't have to add a delay. I know that's tough sometimes. I'll think on it and let you know if I come up with anything by tomorrow morning.

Reply:
I was trying to verify if there was a visual cue for when things are ready. But I don't think I was able to see one. I don't think it is reliant on autosave duration. Maybe it could be the Cloud Environment area in the top right.

Reply:
From what I can tell, waiting for `Kernel Idle` is taking care of making sure the runtime is ready, so that's good. However, when the test is shutting down and trying to close the window, it will display a modal if there are unsaved changes to the notebook. Without the 10 second delay, the modal blocks closing the window which causes the test to fail... and in my local runs, I don't even get a stack trace! With the 10 second delay, the test timing is shifted enough that an auto-save happens between running and closing the window during test tear-down. (I found that the 10 second delay was also effective even if moved to after running the notebook.)

I made a PR (#2569) into this branch with a change that replaces the 10 second delay with a click on the save button. (Since this PR isn't going to dev, it shouldn't have the 2-reviewer requirement. It'll ultimately be reviewed as part of this PR if you choose to merge it to this branch.)

Reply:
Merged your PR. Thanks!